airflow-core/src/airflow/api_fastapi/core_api/routes/public/assets.py

```diff
@@ -182,16 +183,28 @@ def get_assets(
             subqueryload(AssetModel.scheduled_dags),
             subqueryload(AssetModel.producing_tasks),
             subqueryload(AssetModel.consuming_tasks),
+            subqueryload(AssetModel.watchers).joinedload(AssetWatcherModel.trigger),
         )
     )
 
     assets = []
 
     for asset, last_asset_event_id, last_asset_event_timestamp in assets_rows:
+        watchers_data = [
+            {
+                "name": watcher.name,
+                "trigger_id": watcher.trigger_id,
+                "trigger_classpath": watcher.trigger.classpath,
```

Comment:
I'm slightly nervous on general principle about exposing the class path

Reply:
I can remove it. I was only thinking it might be useful information in the UI, but maybe not.

Reply:
I've removed `trigger_classpath` now.